src/js/background.js

```diff
@@ -1043,6 +1043,18 @@ Badger.prototype = {
     }
   },
 
+  // run from a loop of domains given in ./webrequest.js
+  removeDomainWidgetReplacementExceptions: function(domain) {
```

Comment:
We don't really need this function, who else would use it? Let's just do the work directly in that webrequest.js case block. We should end up with fewer lines of code and easier to understand logic (as there will be less of it).